autopilot.go

```diff
@@ -23,39 +25,69 @@ func main() {
 
 type AutopilotPlugin struct{}
 
+func venerableAppName(appName string) string {
+	return fmt.Sprintf("%s-venerable", appName)
+}
+
+func getActionsForExistingApp(appRepo *ApplicationRepo, appName, manifestPath, appPath string) []rewind.Action {
+	return []rewind.Action{
+		// rename
+		{
+			Forward: func() error {
+				return appRepo.RenameApplication(appName, venerableAppName(appName))
+			},
+		},
+		// push
+		{
+			Forward: func() error {
+				return appRepo.PushApplication(manifestPath, appPath)
+			},
+			ReversePrevious: func() error {
+				// If the app cannot start we'll have a lingering application
+				// We delete this application so that the rename can succeed
+				appRepo.DeleteApplication(appName)
```

Comment:
If this fails we should probably return the error so that things stop and the user can fix it manually.